src/pykiso/test_result/assert_step_report.py

```diff
@@ -271,53 +271,51 @@ def func_wrapper(*args, **kwargs):
             # Context
             currentframe = inspect.currentframe()
             f_back = currentframe.f_back
+
             test_name = f_back.f_code.co_name
             test_case_inst: TestCase = assert_method.__self__
             test_class_name = type(test_case_inst).__name__
             assert_name = assert_method.__name__
 
             # filter parent call, only known function recorded
             parent_method = re.findall(_FUNCTION_TO_APPLY, test_name.lower())
-            if parent_method:
-                # get the decorated test fixture name (setUp, tearDown, ...)
-                if parent_method[0] == "handle_interaction":
-                    test_name = f_back.f_locals["func"].__name__
-
-                # Assign variables to signature
-                signature = inspect.signature(assert_method)
-                arguments = signature.bind(*args, **kwargs).arguments
-                test_name = test_case_inst.step_report.current_table or test_name
-
-                # 1. Gather message, var_name, expected, received
-                # 1.1 Get message. default value: ""
-                if test_case_inst.step_report.message:
-                    message = test_case_inst.step_report.message
-                    test_case_inst.step_report.message = ""
-                else:
-                    message = arguments.get("msg", "")
-
-                # ensure message is always present in the arguments
-                # dictionary. (used in _get_expected)
-                if not message and "msg" in signature.parameters:
-                    arguments["msg"] = ""
-
-                # 1.2. Get 'received" value (Always 1st argument)
-                received = list(arguments.values())[0]
-
-                # 1.3. Get variable name
-                var_name = _get_variable_name(f_back, assert_name)
-
-                # 1.4. Get Expected value
-                expected = _get_expected(assert_name, arguments)
-
-                # 2. Update report data
-                # 2.1 Ensure report ready for update
-                _prepare_report(test_case_inst, test_name)
-
-                # 2.2. Add new step
-                _add_step(
-                    test_class_name, test_name, message, var_name, expected, received
-                )
+            # get the decorated test fixture name (setUp, tearDown, ...)
```

Comment:
After discussion, the nested assertions within method calls should all be part of the current running testcase, i.e. when parent_method doesn't start with `test_` then the call stack should be inspected until a valid candidate is found

Reply:
Perfect ,(. I will apply the changes